src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,206 @@
+    //m_elevatorTab.add("pid",m_elevatorPID);
+    //m_elevatorTab.addNumber("",()-> getHeight());
+    m_elevatorPID.setTolerance(0.03);
+    m_elevatorFF = new ElevatorFeedforward(ElevatorConstants.kS, ElevatorConstants.kG, ElevatorConstants.kV, ElevatorConstants.kA);
+
+    m_topLimitSwitch = new DigitalInput(ElevatorConstants.kTopLimitSwitchPort); 
+    m_bottomLimitSwitch = new DigitalInput(ElevatorConstants.kBottomLimitSwitchPort); 
+    addChild("Upper Limit",m_topLimitSwitch);
+    addChild("Bottom Limit",m_bottomLimitSwitch);
+
+
+    //TODO: log, addDouble doesn't work. 
+    //LogManager.addDouble("Elevator/error", () -> {return m_pid.getSetpoint() - getElevatorHeight();});
+    
+    //m_motor.setSafetyEnabled(true);
+    setUpElevatorTab();
+    setSetpoint(0.5); 
+  }
+
+  @Override
+  public void periodic() {
+    /**
+     * If we hit the bottom limit switch, then we must be at the zero position.
+     * Thus set the setpoint to 0, then atSetpoint() will be triggered, causing the 
+     * ResetEncoderAtBottom() command to end
+     */
+    if(m_enabled && m_isCalibrated) {
+      double pid = m_elevatorPID.calculate(getHeight());
+      double ff = m_elevatorFF.calculate(ElevatorConstants.kVelocity, ElevatorConstants.kAccel); 
+      ff = 0; 
+      double pidClamped = MathUtil.clamp(pid, -ElevatorConstants.kPowerLimit, ElevatorConstants.kPowerLimit);
+      double finalMotorPower = pidClamped + ff; 
+      
+      set(finalMotorPower);
+    } else {
+      //m_motor.feed();
+    }
+
+  }
+
+  public void close() {
+    // close the ports
+    //TODO: Is there any way to close the ports of the limit switches attached to the motor? 
+    m_topLimitSwitch.close(); 
+    m_bottomLimitSwitch.close(); 
+  }
+
+  public void set(double power) {
+    if(isBottom()){
+      if(power >0 ){ 
+        m_motor.set(power); 
+      }else{
+        m_motor.set(0); 
+      }
+    }
+
+    else if(isTop()){
+      if(power<0){
+        m_motor.set(power); 
+      }else{
+        m_motor.set(0); 
+      }
+    }
+
+    else{
+      m_motor.set(power); 
+    }
+
+  
+  }
+  
+  public void setEnabled(boolean isEnabled){
+    m_enabled = isEnabled; 
+  }
+
+  public void setIsCalibrated(){
+    m_isCalibrated = true; 
+  }
+
+
+  public void setSetpoint(double setpoint){
+    m_elevatorPID.setSetpoint(setpoint);
+  }
+
+  // public boolean getBottomLimitSwitch(){
+  //  if(m_motor.getSensorCollection().isRevLimitSwitchClosed() == 0){
+  //   return true; 
+  //  }
+  //   return false; 
+  // }
+
+  // public boolean getTopLimitSwitch(){
+  //   if(m_motor.getSensorCollection().isFwdLimitSwitchClosed() == 0){
+  //    return true; 
+  //   }
+  //    return false; 
+  // }
+
+  public boolean isBottom(){
+    if(m_bottomLimitSwitch.get() == false){
+      return true; 
+    } else{
+      return false; 
+    }
+  }
+
+  public boolean isTop(){
+    if(m_topLimitSwitch.get() == false){
+      return true; 
+    } else{
+      return false; 
+    }  }
```

Comment:
please make the method name make a bit more sense like "isAtTop"
just return the invert of the boolean like:
  return !m_limitSwitch.get();

also how did you mess up the brackets so badly

Reply:
I fixed this but leaveing this unresolved so someone sees it

Reply:
ty